library_management_system_downloader/create_api_request.py

```diff
@@ -0,0 +1,51 @@
+import backoff
+from ratelimit import rate_limited
+import requests
+import logging
+
+
+class ErrorWithAPI(Exception):
+    pass
+
+
+# From the backoff documentation (https://pypi.python.org/pypi/backoff),
+# set maximum number of tries on a failed download:
+@backoff.on_exception(
+        backoff.expo,
+        requests.exceptions.RequestException,
```

Comment:
~~Hm I'm not sure why backoff wouldn't be wanted for other exceptions like a `ConnectionError` or `Timeout`? I could even see it sometimes being helpful for any non 200-code response.~~

Now [I see](http://docs.python-requests.org/en/master/_modules/requests/exceptions/) that `RequestException` is a superclass.